### PR TITLE
rd - add collegiateSubreddits table

### LINKED ISSUE #6: Table component for CollegiateSubreddits
Table component for CollegiateSubreddits

Using the `UCSBDatesTable` component as a model, create a similar component for CollegiateSubreddits.

# ACs

- [x] The table has columns for database fields (i.e. the attributes of the corresponding `@Entity` class
- [x] There are fixtures under `/frontend/src/fixtures` for this data type, similar to the other fixtures files.
- [x] There is a storybook entry similar to the one for `UCSBDatesTable`
- [x] 100% coverage with `npm run coverage`
- [x] 100% coverage with `npx stryker run`

### PR DESCRIPTION
**Pull request by rsdharmaji**
# Overview

This PR adds the table for collegiateSubreddits.

# Issues Addressed

resolves #6 

# Details

Table is added along with tests (with same weird code coverage issues as previous table PR).

## Reminders (delete this section before merging or as code is merged)
- [ ] Explicitly link to this pull request each issue that it closes
- [ ] Assign one and ideally two reviewers to review this pull request. 
- [ ] Reviewers should not just approve with "Looks good!" Instead, reviewers should carefully go through the code and use GitHub's reviewer interface to ask questions, make comments, and suggest changes.

### CODE REVIEW

File: frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    ];
+
+    if (hasRole(currentUser, "ROLE_ADMIN")) {
+        columns.push(ButtonColumn("Edit", "primary", editCallback, "CollegiateSubredditsTable"));
+        columns.push(ButtonColumn("Delete", "danger", deleteCallback, "CollegiateSubredditsTable"));
+    } 
+
+    // Stryker disable next-line ArrayDeclaration : [columns] is a performance optimization
+    const memoizedColumns = React.useMemo(() => columns, [columns]);
+    const memoizedDates = React.useMemo(() => subreddits, [subreddits]);
```
Comment (DannyCoco) [inline frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js]: Shouldn't this be 'memoizedSubreddits' and change line 59 when you reference it? I don't think it matters much but just out of uniformity

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import collegiateSubredditsTable from "main/components/collegiateSubreddits/collegiateSubredditsTable";
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: change line to:
import CollegiateSubredditsTable from "main/components/CollegiateSubreddits/CollegiateSubredditsTable";

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { collegiateSubredditsFixtures } from 'fixtures/collegiateSubredditsFixtures';
+
+export default {
+    title: 'components/collegiateSubreddits/collegiateSubredditsTable',
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: Change to:
components/CollegiateSubreddits/CollegiateSubredditsTable

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    component: collegiateSubredditsTable
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: change to:
CollegiateSubredditsTable

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+const Template = (args) => {
+    return (
+        <collegiateSubredditsTable {...args} />
+    )
+};
+
+export const Empty = Template.bind({});
+
+Empty.args = {
+    dates: []
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: change line to: 
subreddits: []

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+export const ThreeDates = Template.bind({});
+
+ThreeDates.args = {
+    dates: collegiateSubredditsFixtures.threeDates
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: change to: 
subreddits: collegiateSubredditsFixtures.threeCollegiateSubreddits

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+export const Empty = Template.bind({});
+
+Empty.args = {
+    dates: []
+};
+
+export const ThreeDates = Template.bind({});
+
+ThreeDates.args = {
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: change to:
ThreeCollegiateSubreddits.args = {

File: frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    )
+};
+
+export const Empty = Template.bind({});
+
+Empty.args = {
+    dates: []
+};
+
+export const ThreeDates = Template.bind({});
```
Comment (DannyCoco) [inline frontend/src/stories/components/CollegiateSubreddits/collegiateSubredditsTable.stories.js]: export const ThreeCollegiateSubreddits = Template.bind({});

File: frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+        </MemoryRouter>
+      </QueryClientProvider>
+
+    );
+
+    await waitFor(() => { expect(getByTestId(`CollegiateSubredditsTable-cell-row-0-col-id`)).toHaveTextContent("1"); });
+
+    const editButton = getByTestId(`CollegiateSubredditsTable-cell-row-0-col-Edit-button`);
+    expect(editButton).toBeInTheDocument();
+
```
Comment (DannyCoco) [inline frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js:115]: Shouldn't you also have: 

const deleteButton = getByTestId(`CollegiateSubredditsTable-cell-row-0-col-Delete-button`);
expect(deleteButton).toBeInTheDocument();

Similar to the given UCSBDates example?

File: frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+      </QueryClientProvider>
+
+    );
+
+    await waitFor(() => { expect(getByTestId(`CollegiateSubredditsTable-cell-row-0-col-id`)).toHaveTextContent("1"); });
+
+    const editButton = getByTestId(`CollegiateSubredditsTable-cell-row-0-col-Edit-button`);
+    expect(editButton).toBeInTheDocument();
+    
+    fireEvent.click(editButton);
```
Comment (DannyCoco) [inline frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js:116]: Shouldn't you add in:
fireEvent.click(deleteButton);

File: frontend/src/tests/utils/collegiateSubredditUtils.test.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+jest.mock('react-toastify', () => {
+    const originalModule = jest.requireActual('react-toastify');
+    return {
+        __esModule: true,
+        ...originalModule,
+        toast: (x) => mockToast(x)
+    };
+});
+
+describe("UCSBSubjectUtils", () => {
```
Comment (DannyCoco) [inline frontend/src/tests/utils/collegiateSubredditUtils.test.js]: change to:
describe("CollegiateSubredditUtils", () => {

File: frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js (hunk 1 above)
Comment (rsdharmaji) [reply frontend/src/main/components/CollegiateSubreddits/CollegiateSubredditsTable.js]: fixed in new commit

File: frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js (hunk 9 above)
Comment (rsdharmaji) [reply frontend/src/tests/components/CollegiateSubreddits/CollegiateSubredditsTable.test.js:115]: it is there, just in the previous test (follows the format of the UCSBDates example).

**Comment by rsdharmaji** [comment]
all comments covered by Danny have been addressed either here or in a new commit.

**Pull request merged**